File: php_payway/src/enumeration.php
<?php

enum ABAPaymentOption
{
    case cards;
    case abapay;
    case abapay_deeplink;
    case bakong;
    case alipay;
    case wechat;
}

 $ABAPaymentOptionMap = [
    "cards" => ABAPaymentOption::cards,
    "abapay" => ABAPaymentOption::abapay,
    "abapay_deeplink" => ABAPaymentOption::abapay_deeplink,
    "bakong" => ABAPaymentOption::bakong,
    "alipay" => ABAPaymentOption::alipay,
    "wechat" => ABAPaymentOption::wechat,
];

/// acceptable currency
enum ABATransactionCurrency
{
    case USD;
    case KHR;
}

$ABATransactionCurrencyMap = [
    'USD' => ABATransactionCurrency::USD,
    'KHR' => ABATransactionCurrency::KHR,
];

/// transaction type
enum ABATransactionType
{
    case purchase;
    case refund;
}

$ABATransactionTypeMap = [
    'purchase' => ABATransactionType::purchase,
    'refund' => ABATransactionType::refund,
];

File: dart_payway/lib/src/enumeration.dart
enum PaywayPaymentOption {
  cards,
  abapay,
  abapay_deeplink,
  abapay_khqr_deeplink,
  bakong,
  alipay,
  wechat
}

const $PaywayPaymentOptionMap = {
  "cards": PaywayPaymentOption.cards,
  "abapay": PaywayPaymentOption.abapay,
  "abapay_deeplink": PaywayPaymentOption.abapay_deeplink,
  "abapay_khqr_deeplink": PaywayPaymentOption.abapay_khqr_deeplink,
  "bakong": PaywayPaymentOption.bakong,
  "alipay": PaywayPaymentOption.alipay,
  "wechat": PaywayPaymentOption.wechat,
};

/// acceptable currency
enum PaywayTransactionCurrency { USD, KHR }

const $PaywayTransactionCurrencyMap = {
  'USD': PaywayTransactionCurrency.USD,
  'KHR': PaywayTransactionCurrency.KHR,
};

/// transaction type
enum PaywayTransactionType { purchase, refund }

const $PaywayTransactionTypeMap = {
  'purchase': PaywayTransactionType.purchase,
  'refund': PaywayTransactionType.refund,
};
